addons/mail/static/src/web/thread_service_patch.js

```diff
@@ -213,7 +213,10 @@ patch(ThreadService.prototype, "mail/web", {
         }
         this._super(...arguments);
     },
-    _openChatWindow(thread, replaceNewMessageChatWindow) {
+    async _openChatWindow(thread, replaceNewMessageChatWindow) {
+        if (thread.type === "chatter") {
+            await this.fetchData(thread, []);
+        }
```

Comment:
With changes in message_format to get thread has access rights, there's no need to actively fetch data of chatter. Let the auto-fetch messages of `Thread` component fetch the messages and receive data of thread access rights in the returned formatted data.